integration/integration_test.go

```diff
@@ -0,0 +1,230 @@
+package integration
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	ps "github.com/mitchellh/go-ps"
+	"github.com/onsi/gomega/gexec"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const TraceeBinaryPath = "../tracee-ebpf/dist/tracee-ebpf"
+
+func checkTraceeBinary(t *testing.T) {
+	if _, err := os.Stat(TraceeBinaryPath); os.IsNotExist(err) {
+		require.FailNow(t, "failed to find tracee binary", err)
+	}
+}
+
+// load tracee into memory with args
+func loadTracee(t *testing.T, w io.Writer, done chan bool, args ...string) {
+	cmd := exec.Command(TraceeBinaryPath, args...)
+	fmt.Println("running: ", cmd.String())
+
+	session, err := gexec.Start(cmd, w, w)
+	require.NoError(t, err)
+	<-done
+	session.Interrupt()
+}
+
+// get pid by process name
+func getPidByName(t *testing.T, name string) int {
+	processes, err := ps.Processes()
+	require.NoError(t, err)
+
+	for _, p := range processes {
+		if strings.Contains(p.Executable(), name) {
+			return p.Pid()
+		}
+	}
+	return -1
+}
+
+// small set of actions to trigger a magic write event
+func checkMagicwrite(t *testing.T, gotOutput *bytes.Buffer) {
+	// create a temp dir for testing
+	d, err := ioutil.TempDir("", "Test_MagicWrite-dir-*")
+	require.NoError(t, err)
+
+	// cp a file to trigger
+	f, err := os.CreateTemp(d, "Test_MagicWrite-file-*")
+	require.NoError(t, err)
+	defer func() {
+		os.Remove(d)
+	}()
+
+	f.WriteString(`foo.bar.baz`)
+	f.Close()
+
+	cpCmd := exec.Command("cp", f.Name(), filepath.Join(d+filepath.Base(f.Name())+"-new"))
+	fmt.Println("executing: ", cpCmd.String())
+	cpCmd.Stdout = os.Stdout
+	assert.NoError(t, cpCmd.Run())
+
+	// check tracee output
+	assert.Contains(t, gotOutput.String(), `bytes: [102 111 111 46 98 97 114 46 98 97 122]`)
+}
+
+// execute a ls command
+func checkExeccommand(t *testing.T, gotOutput *bytes.Buffer) {
+	_, _ = exec.Command("ls").CombinedOutput()
```

Comment:
do we need a sleep between the action and observing it's side affect? (same for other tests)

Reply:
currently the flow is 2 part: 1 - start tracee 2- exercise command an check output. 
a suggestion, to make it 3 phase:
1 - start tracee 2- exercise command and stop tracee 3 - check output.
this will address my previous comment and also possible ease on the test overhead. It might be a premature optimization though

Reply:
I haven't observed any noticeable delay/lag in executing a command and checking its output. The output buffer is shared between tracee and the check function. 

Having said that maybe it will become a problem if we have more test cases but I would defer solving that until it becomes a problem.